pytensor/sparse/basic.py

```diff
@@ -4260,3 +4256,85 @@ def grad(self, inputs, grads):
 
 
 construct_sparse_from_list = ConstructSparseFromList()
+
+
+class SparseBlockDiagonalMatrix(BaseBlockDiagonal):
+    def make_node(self, *matrices, format: Literal["csc", "csr"] = "csc", name=None):
```

Comment:
The format should be part of init and a prop. It should always be possible to do `out.owner.make_node(*out.owner.inputs)` safely, which wouldn't work here.

This is needed when rebuilding graphs with `clone_replace` for example

Reply:
Ok -- I thought I was being clever by delaying the need to commit to a format as late as possible.